contrib/storage-http/src/main/java/org/apache/drill/exec/store/http/HttpAPIConfig.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.drill.exec.store.http;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.drill.common.PlanStringBuilder;
+import org.apache.drill.common.exceptions.UserException;
+import org.apache.parquet.Strings;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Map;
+import java.util.Objects;
+
+public class HttpAPIConfig {
+  private static final Logger logger = LoggerFactory.getLogger(HttpAPIConfig.class);
+
+  private final String url;
+
+  private final String method;
+
+  private final Map<String, String> headers;
+
+  private final String authType;
+
+  private final String userName;
+
+  private final String password;
+
+  private final String postBody;
+
+  public HttpAPIConfig(@JsonProperty("url") String url,
+                       @JsonProperty("method") String method,
+                       @JsonProperty("headers") Map<String, String> headers,
+                       @JsonProperty("authType") String authType,
+                       @JsonProperty("userName") String userName,
+                       @JsonProperty("password") String password,
+                       @JsonProperty("postBody") String postBody) {
+
+    this.headers = headers;
+    this.method = Strings.isNullOrEmpty(method)
+        ? "GET" : method.trim().toUpperCase();
+
+    // Get the request method.  Only accept GET and POST requests.  Anything else will default to GET.
+    switch (this.method) {
+      case "GET":
```

Comment:
Maybe create enum?

Reply:
Converted to enum.

Reply:
Since this is a plugin config, we want to keep types as a simple as possible. Using a string for the method is fine.

An open question is whether validation should occur here or at the point where the information is used. If here, then we'll reject a badly-formed method when the user tries to edit it in the UI. However, we'll also prevent the Drillbit from starting if that bad JSON makes it to persistent storage (or we change the format later on.)

For now, doing validation here is fine until we come up with a uniform solution for all configs.

Reply:
@paul-rogers 
I do think there should be some means of verifying plugin config, ideally with a helpful error message when the user attempts to submit an invalid plugin config.  Right now, unfortunately, all they get is an error message saying somthing like `Invalid JSON` with no real way of determining the cause. 

The syntax highlighter helps if it is an actual syntax error, but for invalid configs, the user really has no way of knowing.  This really isn't for this PR, but something to think about.